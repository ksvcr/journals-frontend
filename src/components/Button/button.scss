.button {
  text-decoration: none;
  outline: 0;
  border-radius: 6px;
  background: #fbfaf5;
  color: $blue-color;
  padding: 18px 25px;
  font-weight: 700;
  line-height: 16px;
  border: 1px solid $gray-color;
  transition: all .2s;
  display: inline-block;
  vertical-align: middle;
  cursor: pointer;

  & + & {
    margin-left: 20px;
  }

  &:hover,
  &:focus {
    border-color: $orange-color;
  }

  &:active,
  &_active {
    background: $orange-color;
  }

  &_small {
    padding: 14px 20px;
  }


  &:disabled {
    background: #e2e2e2;
    border-color: #e2e2e2;
    color: #fff;
  }

  &_orange {
    background: $orange-color;
    border-color: $orange-color;
    color: #fff;


    &:hover,
    &:focus {
      background: #fdc639;
    }

    &:active,
    &_active {
      background: #ff9a00;
    }


    &:disabled {
      background: #e2e2e2;
      border-color: #e2e2e2;
    }
  }

  &_transparent {
    background: transparent;
    border: 0;

    &:hover,
    &:focus {
      background: transparent;
      color: $orange-color;
    }

    &:active,
    &_active {
      background: transparent;
      color: $orange-color;
    }
  }
}
