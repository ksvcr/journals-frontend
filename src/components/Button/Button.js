import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './button.scss';

class Button extends Component {
  handleClick = (event) => {
    const { onClick } = this.props;

    if (onClick) {
      onClick(event);
    }
  };

  render() {
    const { type, href, className, children } = this.props;
    const buttonClasses = classNames('button', className);

    if (type === 'link') {
      return (
        <Link to={ href } className={ buttonClasses }>
          { children }
        </Link>
      );
    } else {
      return (
        <button type={ type } className={ buttonClasses } onClick={ this.handleClick }>
          { children }
        </button>
      );
    }
  }
}

Button.defaultProps = {
  type: 'button',
  href: '/'
};

Button.propTypes = {
  type: PropTypes.string,
  href: PropTypes.string
};

export default Button;
