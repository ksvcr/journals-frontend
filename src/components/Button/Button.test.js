import React from 'react';
import TestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import Button from './Button';

describe('Render Button', () => {
  it('render correctly default Button component', () => {
    const ButtonComponent = TestRenderer.create(<Button>Кнопка</Button>).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });

  it('render correctly Button component with link type', () => {
    const props = {
      type: 'link'
    };

    const ButtonComponent = mount(
      <MemoryRouter>
        <Button { ...props }>Кнопка</Button>
      </MemoryRouter>);

    expect(ButtonComponent.exists('a')).toEqual(
      true
    );
  });
});

