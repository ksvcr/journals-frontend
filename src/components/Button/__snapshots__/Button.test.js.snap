// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button render correctly default Button component 1`] = `
<button
  className="button"
  type="button"
>
  Кнопка
</button>
`;
